include/irr/video/IGPUCommandBuffer.h

```diff
@@ -0,0 +1,264 @@
+    //virtual void blitImage(IGPUImage* srcImage, IGPUImage::E_LAYOUT srcImageLayout, IGPUImage* dstImage, IGPUImage::E_LAYOUT dstImageLayout, uint32_t regionCount, const VkImageBlit* pRegions, IGPUSampler::E_TEXTURE_FILTER filter) = 0;
+    //virtual void resolveImage(IGPUImage* srcImage, IGPUImage::E_LAYOUT srcImageLayout, IGPUImage* dstImage, IGPUImage::E_LAYOUT dstImageLayout, uint32_t regionCount, const VkImageResolve* pRegions) = 0;
+
+    virtual void bindVertexBuffers(uint32_t firstBinding, uint32_t bindingCount, IGPUBuffer** pBuffers, const size_t pOffsets) = 0;
+
+    virtual void setScissor(uint32_t firstScissor, uint32_t scissorCount, const VkRect2D* pScissors) = 0;
+    virtual void setDepthBounds(float minDepthBounds, float maxDepthBounds) = 0;
+    virtual void setStencilCompareMask(E_STENCIL_FACE_FLAGS faceMask, uint32_t compareMask) = 0;
+    virtual void setStencilWriteMask(E_STENCIL_FACE_FLAGS faceMask, uint32_t writeMask) = 0;
+    virtual void setStencilReference(E_STENCIL_FACE_FLAGS faceMask, uint32_t reference) = 0;
+
+    virtual void dispatch(uint32_t groupCountX, uint32_t groupCountY, uint32_t groupCountZ) = 0;
+    virtual void dispatchIndirect(IGPUBuffer* buffer, size_t offset) = 0;
+    virtual void dispatchBase(uint32_t baseGroupX, uint32_t baseGroupY, uint32_t baseGroupZ, uint32_t groupCountX, uint32_t groupCountY, uint32_t groupCountZ) = 0;
+
+    //virtual void setEvent(VkEvent event, VkPipelineStageFlags stageMask) = 0;
+    //virtual void resetEvent(VkEvent event, VkPipelineStageFlags stageMask) = 0;
+
+    //`srcStagemask`, `dstStageMask` take bits from E_PIPELINE_STAGE_FLAGS
+    /*virtual void waitEvents(uint32_t eventCount, const VkEvent* pEvents, std::underlying_type_t<E_PIPELINE_STAGE_FLAGS> srcStageMask, std::underlying_type_t<E_PIPELINE_STAGE_FLAGS> dstStageMask,
+        uint32_t memoryBarrierCount, const VkMemoryBarrier* pMemoryBarriers,
+        uint32_t bufferMemoryBarrierCount, const VkBufferMemoryBarrier* pBufferMemoryBarriers,
+        uint32_t imageMemoryBarrierCount, const VkImageMemoryBarrier* pImageMemoryBarriers
+    ) = 0;*/
+
+    /*virtual void pipelineBarrier(uint32_t eventCount, const VkEvent* pEvents, std::underlying_type_t<E_PIPELINE_STAGE_FLAGS> srcStageMask, std::underlying_type_t<E_PIPELINE_STAGE_FLAGS> dstStageMask,
+        std::underlying_type_t<E_DEPENDENCY_FLAGS> dependencyFlags,
+        uint32_t memoryBarrierCount, const VkMemoryBarrier* pMemoryBarriers,
+        uint32_t bufferMemoryBarrierCount, const VkBufferMemoryBarrier* pBufferMemoryBarriers,
+        uint32_t imageMemoryBarrierCount, const VkImageMemoryBarrier* pImageMemoryBarriers) = 0;*/
+
+    //virtual void beginRenderPass(const VkRenderPassBeginInfo* pRenderPassBegin, VkSubpassContents content) = 0;
+    //virtual void nextSubpass(VkSubpassContents contents) = 0;
+    //virtual void endRenderPass() = 0;
+
+    virtual void setDeviceMask(uint32_t deviceMask) { m_deviceMask = deviceMask; }
```

Comment:
maybe make this fully virtual and hide the member, because the OpenGL backend should completely disregard this.